File: PY/Luciano_Chavarria/Algoritmos_de_Ordenamiento/BubbleSort.py
# -*- coding: cp1252 -*-


def BubbleSort(arr, n):
    if n == 1:
        return
    for i in range(n-1):
        if arr[i] > arr[i + 1]:
            temp = arr[i]
            arr[i] = arr[i + 1]
            arr[i + 1] = temp
    BubbleSort(arr, n-1)
    
    
    
if __name__ == "__main__":
    arr = [4,5,8,9,10,2]
    BubbleSort(arr, len(arr))
    print(*arr)

File: Codigos estudiantes por lenguaje/C++/Bryann Valderrama/Algoritmos de Ordenamiento/bubbleSort.cpp
#include <cstdlib>
#include <bits/stdc++.h>
#define NUM 15
using namespace std;

void bubbleSort(int arr[], int len)
{
    if (len == 1)
    {
        return;
    }
    for (int i = 0; i < len; i++)
    {
        for (int j = 0; j < len - 1; j++)
        {
            if (arr[j] > arr[j + 1])
            {
                int temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
}

int main(int argc, char const *argv[])
{
    int lista[NUM] = {};
    for (int i = 0; i < NUM; i++)
    {
        cin >> lista[i];
    }
    int len = sizeof(lista) / sizeof(lista[0]);
    bubbleSort(lista, len);
    for (int j = 0; j < len; j++)
    {
        cout << lista[j] << " ";
    }
    return 0;
}
